candig/server/datamodel/obo_parser.py

```diff
@@ -77,31 +77,36 @@ def __iter__(self):
         with open(self.obo_file) as fstream:
             rec_curr = None # Stores current GO Term
             typedef_curr = None  # Stores current typedef
-            for lnum, line in enumerate(fstream):
-                # obo lines start with any of: [Term], [Typedef], /^\S+:/, or /^\s*/
-                if self.data_version is None:
-                    self._init_obo_version(line)
-                if line[0:6].lower() == "[term]":
-                    rec_curr = self._init_goterm_ref(rec_curr, "Term", lnum)
-                elif line[0:9].lower() == "[typedef]":
-                    typedef_curr = self._init_typedef(rec_curr, "Typedef", lnum)
-                elif rec_curr is not None or typedef_curr is not None:
-                    line = line.rstrip() # chomp
-                    if ":" in line:
-                        if rec_curr is not None:
-                            self._add_to_ref(rec_curr, line, lnum)
+            lnum = None
+            try:
+                for lnum, line in enumerate(fstream):
+                    # obo lines start with any of: [Term], [Typedef], /^\S+:/, or /^\s*/
+                    if self.data_version is None:
+                        self._init_obo_version(line)
+                    if line[0:6].lower() == "[term]":
+                        rec_curr = self._init_goterm_ref(rec_curr, "Term", lnum)
+                    elif line[0:9].lower() == "[typedef]":
```

Comment:
Just for curiosity, this would fail if the line is not 9 character long, would not it?

Reply:
No, it shouldn't. Even if `line == 'abc'`, doing `line[0:9]` would just return `abc` instead of failing